File: agents/go/internal/tools/write_file.go
package tools

import (
	"fmt"
	"os"
	"path/filepath"
)

// WriteFile writes content to a file, creating parent directories if needed.
func WriteFile(args map[string]interface{}, projectPath string) (string, error) {
	filePath, _ := args["path"].(string)
	content, _ := args["content"].(string)

	fullPath := filepath.Join(projectPath, filePath)

	if err := os.MkdirAll(filepath.Dir(fullPath), 0o755); err != nil {
		return "", fmt.Errorf("create directories: %w", err)
	}

	if err := os.WriteFile(fullPath, []byte(content), 0o644); err != nil {
		return "", fmt.Errorf("write file: %w", err)
	}

	return fmt.Sprintf("Written %d bytes to %s", len(content), filePath), nil
}

File: agents/rust/src/tools/write_file.rs
use std::collections::HashMap;
use std::fs;
use std::path::Path;

pub fn write_file(args: &HashMap<String, serde_json::Value>, project_path: &str) -> String {
    let file_path = args.get("path").and_then(|v| v.as_str()).unwrap_or("");
    let content = args.get("content").and_then(|v| v.as_str()).unwrap_or("");

    let full_path = Path::new(project_path).join(file_path);

    if let Some(parent) = full_path.parent() {
        if let Err(e) = fs::create_dir_all(parent) {
            return format!("Error creating directories: {}", e);
        }
    }

    match fs::write(&full_path, content) {
        Ok(_) => format!("Written {} bytes to {}", content.len(), file_path),
        Err(e) => format!("Error writing file: {}", e),
    }
}
